### DISCUSSION TITLE
About timing sync of Ger video translation

### DISCUSSION THREAD

**Discussion by gosuimbalyndh** [openai/whisper#1888] [Q&A] [answered]
Hail
I'm trying to translate this German movie into English or at least transcribing. I see that the translation or transcribing are truly impressive, nothing to complain. And somehow somewhat, the timing of syncing in my output subtitle file (.vtt, .srt) is quite untimely.  Usually the subtitle displays earlier than the audio/voice. I've attached the video and English .txt subtitle after the translation using -medium model.
There might be a few reasonable causes:
- The video quality is also very low and outdated, it's 720x540 resolution.
- My PC specs is quite low-end: i5 10400f with GTX 1660 Super 5GB VRAM.  That's why I'm gonna use -medium model instead of the largest model since my computer might not handle well enough. Are there many differences between -medium and -large? Or Which model would be the best to use in my case
As a result, Can I make the subtitle timely and well-synced? And with the transcription or translation of that English or German subtitle file, Can I translate that subtitle into other language, e.g. French, Japanese, Malaysian?
Thank you. Really appreciate all of the precious assistances 
https://www.mediafire.com/file/owqx7pt6eyq6h1v/S01_Ger.mkv/file
[S01 Ger.txt](https://github.com/openai/whisper/files/13628655/S01.Ger.txt)

**Comment by glangford** [comment, marked as answer]
> Are there many differences between -medium and -large

There are quality differences which vary by language. It's worth experimenting with different models on some shorter content to see whether the difference is worth the extra time; for some cases medium can be "good enough". Note that the large model will not fit in 5GB VRAM.

> Can I make the subtitle timely and well-synced?

Your best option is to use `--word_timestamps True` when producing subtitles. This gives a significant improvement in synchronization.

> Can I translate that subtitle into other language, e.g. French, Japanese, Malaysian?

whisper will only translate non-English content into English, not into other languages

**Reply by gosuimbalyndh** [reply]
Thank you
There's something I'm unable to understand. If I don't perceive VRAM of GPU here incorrectly, is VRAM something as RTX 4060 8GB GDDR6 or something like that? That's really strange when 3000 series GPU with lower price has higher VRAM than 4000 series GPU with higher price. If the OpenAI Whisper here only notice about the VRAM of GPU. then I must conceive that the more GB of VRAM, the better. As a result, we might need to purchase some GPU with >= 10GB VRAM?

**Reply by glangford** [reply]
I can't give advice on Nvidia hardware unfortunately. That said, for the large model you will need > 10GB VRAM to run on GPU (or you can always run on CPU of course). Colab provides Nvidia T4 instances which can be useful for testing the large model. 

Here is some past discussion on benchmarking in case it is helpful. 

* https://github.com/openai/whisper/discussions/530

**Reply by gosuimbalyndh** [reply]
Thanks
May I ask which platform do you conduct Whisper? I only use CMD from Windows 10. It’s very odd that my output result for largev3 is even worse than medium. The transcribe is usually not in Capitalized letter, not coma…something like that. When I use medium, the result is instantly different. Someone told that using something as Hugging Face or Google Colab is better than CMD. But hell yeh, I know nothing but CMD. Do you suggest, advise me anything?

**Reply by glangford** [reply]
I use Mac locally and Google Colab. 

For models, I would recommend avoiding `large-v3` for now, a number of people have reported problems using it. `medium` or `large-v2` are ok. You can read about some of the questions/problems with `large-v3` here.

* https://github.com/openai/whisper/discussions/1783
